fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapper.kt

```diff
@@ -15,21 +15,81 @@ import org.mozilla.fenix.settings.SupportUtils
 internal fun Collection<OnboardingCardData>.toPageUiData(
     showNotificationPage: Boolean,
     showAddWidgetPage: Boolean,
-): List<OnboardingPageUiData> =
-    filter {
-        when (it.cardType) {
-            OnboardingCardType.NOTIFICATION_PERMISSION -> {
-                it.enabled && showNotificationPage
-            }
-            OnboardingCardType.ADD_SEARCH_WIDGET -> {
-                it.enabled && showAddWidgetPage
+    jexlConditions: Map<String, String>,
+    func: (String) -> Boolean,
+): List<OnboardingPageUiData> {
+    // we are first filtering the cards based on Nimbus configuration
+    return filter { it.shouldDisplayCard(func, jexlConditions) }
+        // we are then filtering again based on device capabilities
+        .filter { it.isCardEnabled(showNotificationPage, showAddWidgetPage) }
+        .sortedBy { it.ordering }
+        .map { it.toPageUiData() }
+}
+
+private fun OnboardingCardData.isCardEnabled(
+    showNotificationPage: Boolean,
+    showAddWidgetPage: Boolean,
+): Boolean =
+    when (cardType) {
+        OnboardingCardType.NOTIFICATION_PERMISSION -> {
+            enabled && showNotificationPage
+        }
+
+        OnboardingCardType.ADD_SEARCH_WIDGET -> {
+            enabled && showAddWidgetPage
+        }
+
+        else -> {
+            enabled
+        }
+    }
+
+/**
+ *  Returns true if the the current Nimbus card fulfills all the conditions in order to display it and false otherwise
+ *
+ *  @param func Function that receives a condition as a [String] and returns its jexl evaluation as a [Boolean].
+ *  @param jexlConditions A <String, String> map containing the Nimbus conditions.
+ *
+ *  @return True - if the card should be displayed and False otherwise.
+ */
```

Comment:
nit:

```suggestion
/**
 *  Determines whether the given [OnboardingCardData] should be displayed.
 *
 *  @param func Function that receives a condition as a [String] and returns its JEXL evaluation as a [Boolean].
 *  @param jexlConditions A <String, String> map containing the Nimbus conditions.
 *
 *  @return True if the card should be displayed, otherwise false.
 */
```